x-pack/plugins/ml/public/embeddables/anomaly_charts/anomaly_charts_react_container.tsx

```diff
@@ -108,6 +126,53 @@ const AnomalyChartsContainer: FC<AnomalyChartsContainerProps> = ({
     error,
   } = useAnomalyChartsData(api, services, chartWidth, severity.val, renderCallbacks);
 
+  const dateFormatTz = useDateFormatTz();
+
+  useEffect(() => {
```

Comment:
If I go to a Dashboard in which I have an anomaly charts after first being in the Anomaly Explorer, the popover isn't appearing. It only appears for me after doing a full refresh of the Dashboard page.

I am seeing these errors on going into the Dashboard page:

<img width="600" alt="Screenshot 2024-06-21 at 15 23 50" src="https://github.com/elastic/kibana/assets/7405507/ffec5162-8efa-4452-a562-4d61af706024">

Reply:
This was a race condition because in some places `mlJobService` was still used via dependency cache. Should be fixed in [e35c568](https://github.com/elastic/kibana/pull/186587/commits/e35c568239360498e33f641e40961fafafeb81e5)

Reply:
I can still reproduce this - when starting in the Anomaly Explorer, loading some anomaly charts, doing a drilldown, and then switching to Dashboard.